### DISCUSSION TITLE
external core / exporting "rendered" templates (js,ect) :? via "export tiddler" ??

### DISCUSSION THREAD

**Discussion by dubiouscript** [TiddlyWiki/TiddlyWiki5#6115] [Q&A]
reading   external-js single-file wiki   https://github.com/Jermolene/TiddlyWiki5/pull/5570

.. https://tiddlywiki.com/#%24%3A%2Fcore%2Fui%2FExportTiddlyWikiCore

after finding "ExportTiddlyWikiCore" and then reading through "#Using the external JavaScript template"

https://tiddlywiki.com/#Using%20the%20external%20JavaScript%20template

i was reminded of reading https://github.com/Jermolene/TiddlyWiki5/pull/5570#issuecomment-892571769
> Actually, the docs in this PR are highly complex and unapproachable for all but the most determined users. Necessarily, because what they're putting across is complex and unintuitive.

as i have previously  found  hacky :goggles:  method to get external js working https://github.com/Jermolene/TiddlyWiki5/issues/4443#issuecomment-617853258

?? ----> is it posible would / could it be more `intuative` to add an option to "export tiddler" that has the same effect as the "ExportTiddlyWikiCore" button ?

eg 
https://tiddlywiki.com/#%24%3A%2Fcore%2Ftemplates%2Ftiddlywiki5.js -> dropdown -> "export tiddler" -> 'tiddlywikicore-5.2.0.js'  

and perhaps some method in the tiddler dropdown ->info linking to  "#Using the external JavaScript template"
... even adding the "ExportTiddlyWikiCore" button to the "info" section

**Comment by pmario** [comment]
I'm not sure, if you did read the whole PR thread. This functionality was close to be removed from the core. About 1/3rd of the functions and docs have been removed, because the info was too prominent. The whole thing is pretty much hidden, because it shouldn't be promoted.

Jeremy pointed out that it is an experiment, to show what can be done. ... The whole templating is tricky but very elegantly implemented, because it needed to be backwards compatible.

I use the external core with my webdav setup. It's a perfect fit there, since the browser caches the core. It will also be a perfect fit for "hypercore" systems (formally know as dat-protocol). This system uses a "copy on write" store, which by default is an "version control system" by design. At the moment it profits very much from saving small content files, because the "deduplication" mechanism isn't implemented yet. 

I'm not 100% sure, what you try to achieve, but I'm 100% sure that we shouldn't try to get more info about it into the core.

**Reply by dubiouscript** [reply]
:laughing: 
> I'm not 100% sure, what you try to achieve, but I'm 100% sure that we shouldn't try to get more info about it into the core.
@pmario 

im simply surgesting that  the button to download/export the rendered output be included in the export tiddler menu of the js template

and musing that if 
the documentation about the architecture of tiddly wiki wrt external-js be linked in the template info of the js template
it would improve  discoverablity of both wiki architecture generally and potential external-js functionality 

...simlar to what you mention hear https://github.com/Jermolene/TiddlyWiki5/discussions/6116

> you probably know, how we could make this fast. ... right?

people looking into this have likely found ( as im shore you know ) & infact do mention `since the browser caches the core`
in terms of page load  caching static external resources is a win 

i dont think this information should be hidden :/
i think the diffrent architectural possibilities offered within  tiddlywiki enviroment should be considered explained and wikified ! :beers:
